src/endpoints/sources.js

```diff
@@ -159,6 +237,206 @@ function pathParts(path = "") {
 }
 
 
+/**
+ * Generate an Express endpoint that sends a dataset or narrative Subresource
+ * determined by the request.
+ *
+ * @param {subresourceExtractor} subresourceExtractor - Function to provide the Subresource instance from the request
+ * @returns {expressEndpointAsync}
+ */
+function sendSubresource(subresourceExtractor) {
+  return async (req, res) => {
```

Comment:
Just to double check, this returned function (`async (req, res) => ...`) is the handler that's called by `nextHandler` (within `dispatchByContentType`). This handler ends up running `proxyResponseBodyFromUpstream` but that's not itself a handler (in the express middleware form of the word)?

Reply:
Correct.